src/client/game.cpp

```diff
@@ -504,6 +505,16 @@ class GameGlobalShaderConstantSetter : public IShaderConstantSetter
 		float clr[4] = {star_color.r, star_color.g, star_color.b, star_color.a};
 		m_star_color.set(clr, services);
 
+		auto ambient_light = m_client->getEnv().getAmbientLight();
```

Comment:
The lighting struct is more like a definition table with fields received from a modder. Actual current color should be saved somewhere in the client env since it is global.

Reply:
Well yes, it also seems to be needed to be able to check whether it has changed. We could also make this client-event-driven, but that would be much more verbose.